### ISSUE TITLE
🐛 Bug Report: Unexpected token 'export'

### ISSUE THREAD

**Issue by dweber019** [backstage/backstage#16897]
### 📜 Description

We have updated from version 1.11.x to 1.12.0.
Additionally we have local plugins created with the command `yarn new --select backend-plugin`. These backend plugins now fail with the error below.
This could be related to https://github.com/backstage/backstage/issues/16542 ?

### 👍 Expected behavior

`yarn dev` runs successfully.

### 👎 Actual Behavior with Screenshots

`yarn dev` errors with

```
[1] Build succeeded
[0] Loaded config from app-config.yaml, app-config.local.yaml
[0] <i> [webpack-dev-server] Project is running at:
[0] <i> [webpack-dev-server] Loopback: http://localhost:3000/, http://[::1]:3000/
[0] <i> [webpack-dev-server] Content not from webpack is served from 'C:\Dev\repos\base-images\backstage\packages\app\public' directory
[0] <i> [webpack-dev-server] 404s will fallback to '/index.html'
[0] <i> [webpack-dev-middleware] wait until bundle finished: /
[1] C:\Dev\repos\base-images\backstage\plugins\adoit-backend\src\index.ts:17
[1] export * from './service/router';
[1] ^^^^^^
[1]
[1] SyntaxError: Unexpected token 'export'
[1]     at internalCompileFunction (node:internal/vm:73:18)
[1]     at wrapSafe (node:internal/modules/cjs/loader:1176:20)
[1]     at Module._compile (node:internal/modules/cjs/loader:1218:27)
[1]     at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
[1]     at Module.load (node:internal/modules/cjs/loader:1117:32)
[1]     at Module._load (node:internal/modules/cjs/loader:958:12)
[1]     at Module.require (node:internal/modules/cjs/loader:1141:19)
[1]     at require (node:internal/modules/cjs/helpers:110:18)
[1]     at @internal/plugin-adoit-backend (C:\Dev\repos\base-images\backstage\packages\backend\dist\main.js:698:18)
[1]     at __webpack_require__ (C:\Dev\repos\base-images\backstage\packages\backend\dist\main.js:727:33)
```

### 👟 Reproduction steps

Create an backend plugin with `yarn new --select backend-plugin` and add it like this https://backstage.io/docs/plugins/backend-plugin#developing-your-backend-plugin.

### 🖥️ Your Environment

OS:   Windows_NT 10.0.19044 - win32/x64
node: v16.14.2
yarn: 1.22.17
cli:  0.22.4 (installed)
backstage:  1.12.0

Dependencies:
  @backstage/app-defaults                                   1.2.1
  @backstage/backend-app-api                                0.4.1
  @backstage/backend-common                                 0.18.3
  @backstage/backend-dev-utils                              0.1.1
  @backstage/backend-plugin-api                             0.5.0
  @backstage/backend-tasks                                  0.5.0
  @backstage/catalog-client                                 1.4.0
  @backstage/catalog-model                                  1.2.1
  @backstage/cli-common                                     0.1.12
  @backstage/cli                                            0.22.4
  @backstage/config-loader                                  1.1.9
  @backstage/config                                         1.0.7
  @backstage/core-app-api                                   1.6.0
  @backstage/core-components                                0.12.5
  @backstage/core-plugin-api                                1.5.0
  @backstage/dev-utils                                      1.0.13
  @backstage/errors                                         1.1.5
  @backstage/eslint-plugin                                  0.1.2
  @backstage/integration-aws-node                           0.1.2
  @backstage/integration-react                              1.1.11
  @backstage/integration                                    1.4.3
  @backstage/plugin-adr-backend                             0.3.1
  @backstage/plugin-adr-common                              0.2.7
  @backstage/plugin-adr                                     0.4.1
  @backstage/plugin-api-docs                                0.9.1
  @backstage/plugin-app-backend                             0.3.43
  @backstage/plugin-auth-backend                            0.18.1
  @backstage/plugin-auth-node                               0.2.12
  @backstage/plugin-badges-backend                          0.1.37
  @backstage/plugin-badges                                  0.2.40
  @backstage/plugin-bitbucket-cloud-common                  0.2.4
  @backstage/plugin-catalog-backend-module-bitbucket-server 0.1.8
  @backstage/plugin-catalog-backend-module-bitbucket        0.2.10
  @backstage/plugin-catalog-backend-module-ldap             0.5.10
  @backstage/plugin-catalog-backend                         1.8.0
  @backstage/plugin-catalog-common                          1.0.12
  @backstage/plugin-catalog-graph                           0.2.28
  @backstage/plugin-catalog-import                          0.9.6
  @backstage/plugin-catalog-node                            1.3.4
  @backstage/plugin-catalog-react                           1.4.0
  @backstage/plugin-catalog                                 1.9.0
  @backstage/plugin-github-actions                          0.5.16
  @backstage/plugin-home                                    0.4.32
  @backstage/plugin-jenkins-backend                         0.1.33
  @backstage/plugin-jenkins-common                          0.1.14
  @backstage/plugin-jenkins                                 0.7.15
  @backstage/plugin-kubernetes-backend                      0.9.4
  @backstage/plugin-kubernetes-common                       0.6.1
  @backstage/plugin-kubernetes                              0.7.9
  @backstage/plugin-linguist-backend                        0.2.0
  @backstage/plugin-linguist-common                         0.1.0
  @backstage/plugin-linguist                                0.1.1
  @backstage/plugin-org                                     0.6.6
  @backstage/plugin-permission-common                       0.7.4
  @backstage/plugin-permission-node                         0.7.6
  @backstage/plugin-permission-react                        0.4.11
  @backstage/plugin-proxy-backend                           0.2.37
  @backstage/plugin-scaffolder-backend                      1.12.0
  @backstage/plugin-scaffolder-common                       1.2.6
  @backstage/plugin-scaffolder-node                         0.1.1
  @backstage/plugin-scaffolder-react                        1.2.0
  @backstage/plugin-scaffolder                              1.12.0
  @backstage/plugin-search-backend-module-elasticsearch     1.1.4
  @backstage/plugin-search-backend-module-pg                0.5.4
  @backstage/plugin-search-backend-node                     1.1.4
  @backstage/plugin-search-backend                          1.2.4
  @backstage/plugin-search-common                           1.2.2
  @backstage/plugin-search-react                            1.5.1
  @backstage/plugin-search                                  1.1.1
  @backstage/plugin-shortcuts                               0.3.8
  @backstage/plugin-sonarqube-backend                       0.1.8
  @backstage/plugin-sonarqube-react                         0.1.4
  @backstage/plugin-sonarqube                               0.6.5
  @backstage/plugin-tech-insights-backend-module-jsonfc     0.1.27
  @backstage/plugin-tech-insights-backend                   0.5.9
  @backstage/plugin-tech-insights-common                    0.2.10
  @backstage/plugin-tech-insights-node                      0.4.1
  @backstage/plugin-tech-insights                           0.3.8
  @backstage/plugin-tech-radar                              0.6.2
  @backstage/plugin-techdocs-backend                        1.6.0
  @backstage/plugin-techdocs-module-addons-contrib          1.0.11
  @backstage/plugin-techdocs-node                           1.6.0
  @backstage/plugin-techdocs-react                          1.1.4
  @backstage/plugin-techdocs                                1.6.0
  @backstage/plugin-user-settings                           0.7.1
  @backstage/release-manifests                              0.0.9
  @backstage/test-utils                                     1.2.6
  @backstage/theme                                          0.2.18
  @backstage/types                                          1.0.2
  @backstage/version-bridge                                 1.0.3
Done in 1.76s.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by dweber019** [comment]
Deleted the `yarn.lock` and removed node_modules but no success.

**Comment by dweber019** [comment]
If I revert this change locally it will work again for me https://github.com/backstage/backstage/pull/16548/files#diff-de40b39bd698be14c4aeaf0dcd1632014902ef074c355630cce0809205e15c2d

**Comment by dweber019** [comment]
The reason is that we use `path.join(...)` which is an issue for windows users as the path are rewritten to this
```
[1] [
[1]   'app',
[1]   'backend',
[1]   '@internal\\plugin-adoit',
[1]   '@internal\\plugin-adoit-backend'
[1] ]
```
instead of
```
[1] [
[1]   'app',
[1]   'backend',
[1]   '@internal/plugin-adoit',
[1]   '@internal/plugin-adoit-backend'
[1] ]
```

Fix is simple we should use `path.posix.join(p.packageJson.name, e.mount)` instead of `path.join(p.packageJson.name, e.mount)` else windows implementation will be used which is not compatible with webpack.

Will create a PR.

@Rugvip can this be a hot fix, as 1.12.x update broke all windows based development with @internal plugins.
